drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_rtio.c

```diff
@@ -0,0 +1,127 @@
+/* ST Microelectronics LSM6DSV16X 6-axis IMU sensor driver
+ *
+ * Copyright (c) 2023 Google LLC
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/sensor.h>
+#include "lsm6dsv16x.h"
+#include "lsm6dsv16x_rtio.h"
+#include "lsm6dsv16x_decoder.h"
+#include <zephyr/rtio/work.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LSM6DSV16X_RTIO, CONFIG_SENSOR_LOG_LEVEL);
+
+static void lsm6dsv16x_submit_sample(const struct device *dev, struct rtio_iodev_sqe *iodev_sqe)
+{
+	const struct sensor_read_config *cfg = iodev_sqe->sqe.iodev->data;
+	const struct sensor_chan_spec *const channels = cfg->channels;
+	const size_t num_channels = cfg->count;
+	uint32_t min_buf_len = sizeof(struct lsm6dsv16x_rtio_data);
+	int rc;
+	uint8_t *buf;
+	uint32_t buf_len;
+	struct lsm6dsv16x_rtio_data *edata;
+	struct lsm6dsv16x_data *data = dev->data;
+
+	const struct lsm6dsv16x_config *config = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&config->ctx;
+
+	/* Get the buffer for the frame, it may be allocated dynamically by the rtio context */
+	rc = rtio_sqe_rx_buf(iodev_sqe, min_buf_len, min_buf_len, &buf, &buf_len);
+	if (rc != 0) {
+		LOG_ERR("Failed to get a read buffer of size %u bytes", min_buf_len);
+		rtio_iodev_sqe_err(iodev_sqe, rc);
+		return;
+	}
+
+	edata = (struct lsm6dsv16x_rtio_data *)buf;
+
+	edata->has_accel = 0;
+	edata->has_gyro = 0;
+	edata->has_temp = 0;
+
+	for (int i = 0; i < num_channels; i++) {
+		switch (channels[i].chan_type) {
```

Comment:
do you want to support `SENSOR_CHAN_ALL`?
```c
		case SENSOR_CHAN_ALL:
            edata->has_accel = 1;
            edata->has_gyro = 1;
#if defined(CONFIG_LSM6DSV16X_ENABLE_TEMP)
            edata->has_temp = 1;
#endif
			break;
```

Reply:
I also reviewed the error checking and sqe error reporting